openapi.yml

```diff
@@ -0,0 +1,543 @@
+        '500':
+          description: Unable to get content diff
+      parameters:
+        - name: id
+          in: path
+          required: true
+          example: 'druid:bc123df4567'
+          schema:
+            $ref: '#/components/schemas/Druid'
+      requestBody:
+        required: true
+        content:
+          application/x-www-form-urlencoded:
+            schema:
+              type: object
+              properties:
+                content_metadata:
+                  type: string
+                  description: an XML representation of content metadata
+                version:
+                  example: '9'
+                  type: string
+                subset:
+                  example: 'publish'
+                  description: "subset of files to diff (default: 'all')"
+                  type: string
+                  enum:
+                    - all
+                    - shelve
+                    - publish
+                    - preserve
+              required:
+                - content_metadata
+components:
+  requestBodies:
+    CatalogOptions:
+      description: Common options for PUT/POST/PATCH operations to the catalog controller
+      required: true
+      content:
+        application/x-www-form-urlencoded:
+          schema:
+            $ref: '#/components/schemas/CatalogOptions'
+    CatalogOptionsWithDruid:
+      description: Common options (plus druid) for PUT/POST/PATCH operations to the catalog controller
+      required: true
+      content:
+        application/x-www-form-urlencoded:
+          schema:
+            allOf:
+              - $ref: '#/components/schemas/CatalogOptions'
+              - type: object
+                properties:
+                  druid:
+                    $ref: '#/components/schemas/Druid'
+                required:
+                  - druid
+  schemas:
+    Druid:
+      description: Digital Repository Unique Identifier (bare or prefixed)
+      type: string
+      pattern: '^(druid:)?[b-df-hjkmnp-tv-z]{2}[0-9]{3}[b-df-hjkmnp-tv-z]{2}[0-9]{4}$'
+      example: 'druid:bc123df4567'
+    DruidWithOptionalFormat:
+      description: Digital Repository Unique Identifier (bare or prefixed), with a file extension
+      type: string
+      pattern: '^(druid:)?[b-df-hjkmnp-tv-z]{2}[0-9]{3}[b-df-hjkmnp-tv-z]{2}[0-9]{4}(\.json)?$'
+      example: 'druid:bc123df4567.json'
+    CatalogOptions:
+      description: Options for catalog controller endpoint
+      type: object
+      properties:
+        incoming_version:
+          type: integer
+        incoming_size:
+          type: integer
+        storage_location:
+          description: Must correspond to a real MoabStorageRoot
+          type: string
+        checksums_validated:
+          type: string
+          enum:
+            - 'true'
+            - 'True'
+            - 'TRUE'
+            - 'nil'
+            - '1'
+            - 'on'
```

Comment:
These cases are from https://github.com/sul-dlss/preservation_catalog/blob/master/spec/controllers/catalog_controller_spec.rb#L297-L308 so, yeah, I'm inclined not to tweak the openapi since these are cases we're testing for.

Reply:
ah!  ok.  that totally makes sense.  i hadn't realized we were testing for those odd cases, but i would agree this is a totally reasonable solution.  i think i'd slightly rather that we make the spec more restrictive and then have our tests see `'nil', '1', 'on'` as invalid input instead of falsey options, but that strikes me as a backlog ticket.  don't have strong feelings on that, and certainly wouldn't hold up this PR over that.